core/src/message/context.rs

```diff
@@ -211,8 +210,7 @@ impl ContextStore {
 /// Context of currently processing incoming message.
 #[derive(Debug)]
 pub struct MessageContext {
-    kind: DispatchKind,
-    current: IncomingMessage,
+    dispatch: IncomingDispatch,
```

Comment:
actually it seems bad to use not de-constructed IncomingDispatch since it provides option to misuse which store: MessageContext::store or MessageContext::dispatch::context

Reply:
worth checking if we could remove one of them

Reply:
Yeah, I think with the cloned dispatch it's better to put it back the way it was